code-challenges.test.js

```diff
@@ -20,8 +20,37 @@ const fibonacciLength1 = 6
 const fibonacciLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+describe("genFib", () => {
+  // "genFib" is the string argument for the describe function
+  it("takes in a number (greater than 2) and returns an array containing the Fibonacci sequence", () => {
+  // the "it" statement explains that the function should give us the Fibonacci sequence
+    expect(genFib(fibonacciLength1)).toEqual([1, 1, 2, 3, 5, 8])
+    expect(genFib(fibonacciLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+  // two expect statements are used to call the function for each variable and compare the output  
+  })
+}) // FAIL: ReferenceError: genFib is not defined
+
 // b) Create the function that makes the test pass.
 
+const genFib = (length) => {
+  // created a function called genFib that uses the parameter length as an argument
+  const fibLength = [1, 1]; 
+  // fibLength is an array for the beginning sequence of index 0 & 1 of the Fibonacci
+  for (let i = 2; i < length; i++) {
+  // in the for loop, i begins at index 2 since fibLength is holding index 0 and 1. the index is less than the length of the array and goes to the next index in the iteration until i is less than the length
+    const nextFib = fibLength[i - 1] + fibLength[i - 2];
+  // nextFib holds the value of fibLength[i - 1] + fibLength[i - 2]. since i begins at index 2, fibLength[2 - 1] + fibLength[2 - 2] becomes fibLength[1] + fibLength[0]. then we add the values at those indexes which is 1 + 1 = 2 which is the third value at index 2. when i = 3, its fibLength[2] + fibLength[1], 2 + 1 = 3, the fourth value at index 3. so forth and so on. 
+    fibLength.push(nextFib);
+  // fibLength.push add the value to the end of the array in nextFib and runs it again until the iteration is complete
+  }
+
+  return fibLength;
+  // fibLength is returned with the the completed Fibonacci sequence
+}
+
+// console.log(genFib(fibonacciLength1))
+// console.log(genFib(fibonacciLength2))
+
```

Comment:
Beautiful! Remember that your test will call your function. The console logs will likely cause an error.